### PR TITLE
docs: Rust client usage to /guides

### PR DESCRIPTION
**Pull request by Anush008**
Following https://github.com/qdrant/landing_page/pull/401, this PR adds the Rust client usage to the guides section of the documentation. Thereby making Rust examples available throughout the docs.

### CODE REVIEW

**Comment by Anush008** [comment]
@timvisee, @generall review 🙏.

File: qdrant-landing/content/documentation/guides/optimize.md [hunk 1]
```diff
@@ -267,6 +364,44 @@ client.createCollection("{collection_name}", {
...
+        vectors_config: Some(VectorsConfig {
+            config: Some(Config::Params(VectorParams {
+                size: 768,
+                distance: Distance::Cosine.into(),
+                ..Default::default()
+            })),
+        }),
+        optimizers_config: Some(OptimizersConfigDiff {
+            memmap_threshold: Some(20000),
+              ..Default::default()
```
Comment (timvisee) [inline qdrant-landing/content/documentation/guides/optimize.md:391]:
```suggestion
            ..Default::default()
```

File: qdrant-landing/content/documentation/guides/quantization.md [hunk 2]
```diff
@@ -276,6 +309,37 @@ client.createCollection("{collection_name}", {
...
+        vectors_config: Some(VectorsConfig {
+            config: Some(Config::Params(VectorParams {
+                size: 1536,
+                distance: Distance::Cosine.into(),
+                ..Default::default()
+            })),
+        }),
+        quantization_config: Some(QuantizationConfig {
+            quantization: Some(Quantization::Binary(BinaryQuantization {
+                    always_ram: Some(true),
```
Comment (timvisee) [inline qdrant-landing/content/documentation/guides/quantization.md:335]:
```suggestion
                always_ram: Some(true),
```

**Review by timvisee — COMMENTED** [review]
@Anush008 Thanks again Anush!

Question: did you reformat all these snippets with `rust fmt` already? In that case I don't have to do it for all of them again.

**Comment by Anush008** [comment]
@timvisee, I did. I think the off ones are those where I tried to make changes after formatting.

I'll recheck again.

**Comment by timvisee** [comment]
> I'll recheck again.

No problem, I think all is fine.

There is a merge conflict that needs to be resolved now though.

**Comment by Anush008** [comment]
> There is a merge conflict that needs to be resolved now though.

Resolved.

**Review by timvisee — APPROVED** [review]

**Comment by timvisee** [comment]
Thanks!

I think it's fine like this. Feel free to merge.

**Pull request merged**